### PR TITLE
Updated email notifications and the email sender class.

### PR DESCRIPTION
**Pull request by Christopher-Tihor**

### CODE REVIEW

File: evacuees/src/API/EMBC.Registrants.API/ProfilesModule/ProfileManager.cs [hunk 1]
```diff
@@ -75,6 +78,20 @@ public async Task<string> SaveProfile(Profile profile)
...
+
+                if (profile != null && profile.ContactDetails.Email != null)
+                {
+                    // Send email notification of new registrant record created
+                    EmailAddress registrantEmailAddress = new EmailAddress
+                    {
+                        Name = profile.PersonalDetails.FirstName + " " + profile.PersonalDetails.LastName,
+                        Address = profile.ContactDetails.Email
+                    };
+                    SendRegistrationNotificationEmail(registrantEmailAddress);
```
Comment (ytqsl) [inline evacuees/src/API/EMBC.Registrants.API/ProfilesModule/ProfileManager.cs:93]: Wouldn't this send the email notification for every update to the profile?

File: evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs [hunk 2]
```diff
@@ -145,14 +147,14 @@ public void Send(EmailMessage emailMessage)
 
                 try
                 {
-                    if (emailConfiguration.SmtpServer == BcGovSmtpServer)
+                    if (emailConfiguration.EnableSSL == false)
```
Comment (ytqsl) [inline evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs:150]: change to a single trinary if

File: evacuees/src/API/EMBC.Tests.Integration.Registrants.API/Evacuations/EvacuationTests.cs [hunk 3]
```diff
@@ -79,6 +79,8 @@ public async Task CanCreateEvacuation()
             var createdEvacuationFile = await evacuationManager.GetEvacuation(testUserId, newEssFileNumber);
 
             createdEvacuationFile.ShouldNotBeNull();
+
+            await evacuationManager.DeleteEvacuation(testUserId, newEssFileNumber);
```
Comment (ytqsl) [inline evacuees/src/API/EMBC.Tests.Integration.Registrants.API/Evacuations/EvacuationTests.cs:83]: Shouldly has the ability to test for exceptions, or check that the evacuation is no longer accessible

**Review by ytqsl — CHANGES_REQUESTED** [review]
All the email templates need to move to files and the code need to move to a different responsible class

File: evacuees/src/API/EMBC.Registrants.API/ProfilesModule/ProfileManager.cs (hunk 1 above)
Comment (Christopher-Tihor) [reply evacuees/src/API/EMBC.Registrants.API/ProfilesModule/ProfileManager.cs:93]: No, it should only send the email notification when a profile is created.

File: evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs (hunk 2 above)
Comment (Christopher-Tihor) [reply evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs:150]: I couldn't find a way to do this in a way that worked and made sense to me. Perhaps you can show me what you had in mind.

Comment (ytqsl) [reply evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs:150]: Sure, let me know when you're available for live share

File: evacuees/src/API/EMBC.Registrants.API/RegistrationsModule/API.cs [hunk 4]
```diff
@@ -186,25 +61,16 @@ public async Task<ActionResult<List<RegistrantEvacuation>>> GetRegistrantEvacuat
     public class AnonymousRegistration
     {
         [Required]
-        public Registration RegistrationDetails { get; set; }
+        public Profile RegistrationDetails { get; set; }
```
Comment (ytqsl) [inline evacuees/src/API/EMBC.Registrants.API/RegistrationsModule/API.cs:64]: Let's discuss this

File: evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs [hunk 5]
```diff
@@ -38,52 +39,25 @@ public class EmailAddress
...
-        {
-            fromAddresses = value;
-        }
-
         public EmailMessage(List<EmailAddress> to, string subject, string body) : this(null, to, subject, body)
-        {
-        }
+        { }
 
         public EmailMessage(List<EmailAddress> fromAddresses, List<EmailAddress> toAddresses, string subject, string body)
```
Comment (ytqsl) [inline evacuees/src/API/EMBC.Registrants.API/Utils/EmailSender.cs:50]: List->IEnumerable

**Review by ytqsl — APPROVED** [review]
Looks good

**Pull request merged**